crates/astria-sequencer/src/transaction/unsigned.rs

```diff
@@ -0,0 +1,118 @@
+use astria_proto::sequencer::v1::{
+    unsigned_transaction::Value::AccountsTransaction as ProtoAccountsTransaction,
+    UnsignedTransaction as ProtoUnsignedTransaction,
+};
+use prost::Message as _;
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+use crate::{
+    accounts::transaction::Transaction as AccountsTransaction,
+    crypto::SigningKey,
+    hash,
+    transaction::signed::Transaction as SignedTransaction,
+};
+
+/// Represents an unsigned sequencer chain transaction.
+/// This type wraps all the different module-specific transactions.
+/// If a new transaction type is added, it should be added to this enum.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[non_exhaustive]
+pub(crate) enum Transaction {
+    AccountsTransaction(AccountsTransaction),
+}
+
+impl Transaction {
+    /// Converts the transaction into its protobuf representation.
+    #[must_use]
+    pub(crate) fn to_proto(&self) -> ProtoUnsignedTransaction {
+        match &self {
+            Transaction::AccountsTransaction(tx) => ProtoUnsignedTransaction {
+                value: Some(ProtoAccountsTransaction(tx.to_proto())),
+            },
+        }
+    }
+
+    /// Signs the transaction with the given signing key.
+    #[allow(dead_code)]
+    #[must_use]
+    pub(crate) fn sign(self, secret_key: &SigningKey) -> SignedTransaction {
```

Comment:
For transformations like convention is usually to call this `to_signed` or smth like that. But because this takes n argument maybe?
```suggestion
    pub(crate) fn to_signed_with_key(self, secret_key: &SigningKey) -> SignedTransaction {
```

Reply:
hmm, this gives me a clippy warning:
```
error: methods with the following characteristics: (`to_*` and `self` type is not `Copy`) usually take `self` by reference
  --> crates/astria-sequencer/src/transaction/unsigned.rs:42:29
   |
42 |     pub(crate) fn to_signed(self, secret_key: &SigningKey) -> Signed {
   |                             ^^^^
   |
   = help: consider choosing a less ambiguous name
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention
   = note: `-D clippy::wrong-self-convention` implied by `-D warnings`
```

Reply:
will change to `into_signed`